libraries/botbuilder-dialogs/src/componentDialog.ts

```diff
@@ -77,7 +93,19 @@ export class ComponentDialog<O extends object = {}> extends Dialog<O> {
      * This defaults to the ID of the first child dialog added using [addDialog()](#adddialog).
      */
     protected initialDialogId: string;
-    private dialogs: DialogSet = new DialogSet(null);
+    private readonly dialogs: DialogSet = new DialogSet(null);
```

Comment:
>readonly [](start = 12, length = 8)

Why make these readonly?  This means you can only set initialize from the constructor?  Makes it non-declaritive